test/userController.test.js

```diff
@@ -94,3 +94,143 @@ describe('GET /api/v1/users/:user_id endpoint', () => {
+      .set('Cookie', [signedCookie]);
+
+    // Assert: Check the response status and user information
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('user');
+    expect(response.body.user).toHaveProperty('name', 'ava');
+    expect(response.body.user).toHaveProperty('email', 'ava@ava.com');
+    expect(response.body.user).toHaveProperty('role', 'user');
+    expect(response.body.user).toHaveProperty('purchasedAlbums');
+    expect(Array.isArray(response.body.user.purchasedAlbums)).toBe(true);
+
+    // Check that each purchased album has required properties
+    response.body.user.purchasedAlbums.forEach((album) => {
+      expect(album).toHaveProperty('title');
+      expect(album).toHaveProperty('artist');
+      expect(album).toHaveProperty('year');
+      // Add more checks for album properties if needed
+    });
+
+    expect(response.body.user).not.toHaveProperty('password');
+  });
+
+  it('should return a 401 status if the user is not authenticated', async () => {
+    // Act: Attempt to get the current user's information with purchased albums without authentication
+    const response = await request(app).get(
+      '/api/v1/users/getCurrentUserWithPurchasedAlbums'
+    );
+
+    // Assert: Check the response status for 401
+    expect(response.status).toBe(401);
+  });
+});
+
+describe('GET /api/v1/users/showCurrentUser endpoint', () => {
+  it('should return the data of the currently authenticated user', async () => {
+    // Act: Log in and get a signed cookie
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Act: Get the current user's information using the authenticated cookie
+    const response = await request(app)
+      .get('/api/v1/users/showCurrentUser')
+      .set('Cookie', [signedCookie]);
+
+    // Assert: Check the response status and user information
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('user');
+    expect(response.body.user).toMatchObject({
+      name: 'ava',
+      email: 'ava@ava.com',
+      role: 'user',
+    });
+    expect(response.body.user).not.toHaveProperty('password');
+  });
+
+  it('should return a 401 status if the user is not authenticated', async () => {
+    // Act: Attempt to get the current user's information without authentication
+    const response = await request(app).get('/api/v1/users/showCurrentUser');
+
+    // Assert: Check the response status for 401
+    expect(response.status).toBe(401);
+  });
+});
+
+describe('GET /api/v1/users endpoint', () => {
+  it('should return a list of users without the password field', async () => {
+    // Act: Log in and get a signed cookie
+    const signedCookie = await loginAndReturnCookie(testUserCredentials);
+
+    // Act: Get a list of users using the authenticated cookie
+    const response = await request(app)
+      .get('/api/v1/users')
+      .set('Cookie', [signedCookie]);
+
```

Comment:
This endpoint only works for admin users. But the `testUserCredentials` are associated with a user, Ava Smith, with the role `'user'`.

We can follow similar steps as what was done to create `Ava` to create an admin user. If we want this admin user to be reusable throughout the tests, then the best place to create it is in the `beforeAll` callback/hook which runs one time when this file is loaded by Jest, before any tests have been run:

1.  In the top of the file, where other variables are declared, declare a new empty variable for `testAdmin`
2.  Declare a `const` variable for `testAdminCredentials` so that we can easily reuse the email and password for loggging in as an admin
3.  In `beforeAll` , use `await createSingleUser` to create a user with the `testAdminCredentials` email and password, select a name and username, and set the `role` to `'admin'`

Then in this test, on line 208, instead of logging in with the `testUserCredentials`, you would want to log in as an admin:

```
    // Arrange: Log in as an admin and get a signed cookie
    const signedCookie = await loginAndReturnCookie(testAdminCredentials);
```